Memorize/MemoryGame.swift

```diff
@@ -24,9 +24,25 @@ struct MemoryGame<CardContent> {
     
     mutating func choose(_ card: Card) {
 //        if let chosenIndex = index(of: card) {
-        if let chosenIndex = cards.firstIndex(where: {$0.id == card.id}) {
+        if let chosenIndex = cards.firstIndex(where: {$0.id == card.id}),
+           !cards[chosenIndex].isFaceUp,
+           !cards[chosenIndex].isMatched
+        {
+            if let potentialMatchIndex = indexOfTheOneAndOnlyFaceUpCard {
```

Comment:
If optional can be unwrapped, it means there is a card that is face up. Might want to make this logic clearer since it involves a lot of visual parsing.